xdsl/pattern_rewriter.py

```diff
@@ -91,6 +91,11 @@ class PatternRewriter(PatternRewriterListener):
     has_done_action: bool = field(default=False, init=False)
```

Comment:
```suggestion
    has_done_action: bool
```
I think `field` is dead code if you add an init?

Reply:
Nope, the `default` still get added. I removed the `init=False` though, as we don't have an `__init__` here.